checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -448,18 +443,13 @@ private boolean hasOrderNonDetListAsTypeArgument(AnnotatedTypeMirror atm) {
         return false;
     }
 
-    /**
-     * Returns true if {@code javaType} may be annotated as {@code @OrderNonDet}.
-     *
-     * @param javaType the type to be checked
-     * @return true if {@code javaType} is Collection (or a subtype), Iterator (or a subtype), or an
-     *     array
-     */
-    public boolean mayBeOrderNonDet(AnnotatedTypeMirror javaType) {
-        return (javaType.getKind() == TypeKind.ARRAY
-                || isCollection(javaType)
-                || isIterator(javaType)
-                || isMap(javaType));
+    /** Returns true if the underlying type of {@code atm} is a collection type. */
+    public boolean isCollectionType(AnnotatedTypeMirror atm) {
+        if (atm.getKind() == TypeKind.ARRAY) {
+            return true;
+        }
+        TypeElement typeElement = TypesUtils.getTypeElement(atm.getErased().getUnderlyingType());
+        return getDeclAnnotation(typeElement, CollectionType.class) != null;
```

Comment:
Should this also return true if any supertype or superinterface is a collection type?  That is what I would expect, and it is how the old code worked.  Otherwise we need many more `@CollectionType` annotations, for example on `ArrayList` and user-defined classes.